cpp/src/arrow/compute/exec/filter_node.cc

```diff
@@ -99,11 +105,35 @@ class FilterNode : public ExecNode {
   void InputReceived(ExecNode* input, ExecBatch batch) override {
     DCHECK_EQ(input, inputs_[0]);
 
-    auto maybe_filtered = DoFilter(std::move(batch));
-    if (ErrorIfNotOk(maybe_filtered.status())) return;
+    ARROW_LOG(DEBUG) << "FilterNode: >> input";
+
+    auto executor = plan()->exec_context()->executor();
```

Comment:
Is the purpose of the logic here to allow the FilterNode to still be used in the old way while allowing it schedule a task as well?

Reply:
Well executor is optional so I think will be maintained to run without executor.

Reply:
While that's currently true, I think we want to require an executor in the future (even if the executor is "block and do the work on the calling thread"). See https://github.com/apache/arrow/pull/10845#issuecomment-904986082 and ARROW-13741.